paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -310,6 +327,9 @@ class Paparazzi @JvmOverloads constructor(
             }
 
             val image = bridgeRenderSession.image
+            if (accessibilityCheckConfig.shouldValidate && renderExtensions.isEmpty()) {
+              validateLayoutAccessibility(modifiedView, image)
+            }
```

Comment:
Yeah that is the reason for the check for no render extensions here. The view does have to be attached to the window before the `LayoutValidator` can successfully build the hierarchy it needs internally to run it's checks. 

I suppose one approach is that we could add the unmodified view to its parent, perform the validations, then remove it and swap in the modified view for the actual snapshotting. This would however result in losing the chance to pass the `image` into the validation as well (the `image` is used for checks such as text contrast to perform a color comparison). 

Happy to try out any other approaches you might have in mind here!